### PR TITLE
Fall back to CPU for Delta Lake metadata queries [databricks]

### PR DESCRIPTION
**Pull request by andygrove**
Closes https://github.com/NVIDIA/spark-rapids/issues/5624

This PR adds a very specific "is this a Delta Lake metadata query?" check and then falls back to CPU in this case.

The integration test currently only runs on Databricks. I filed https://github.com/NVIDIA/spark-rapids/issues/5966 to expand Delta Lake testing to other Spark versions.

### CODE REVIEW

**Comment by abellina** [comment]
Should this fallback be under a config that can be disabled? In case someone does have a table named `_delta_log`, but it is not a delta lake table.

**Comment by andygrove** [comment]
> Should this fallback be under a config that can be disabled? In case someone does have a table named `_delta_log`, but it is not a delta lake table.

Yes, that makes sense. I will add that.

**Comment by andygrove** [comment]
build

File: integration_tests/src/main/python/delta_lake_test.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from asserts import assert_gpu_fallback_collect
+from marks import allow_non_gpu, delta_lake
+from spark_session import with_cpu_session, is_databricks91_or_later
+
+_conf = {'spark.rapids.sql.explain': 'ALL'}
```
Comment (tgravescs) [inline integration_tests/src/main/python/delta_lake_test.py:20]: do we need this on for the test to work properly or just for user to double check output?  I'm assuming latter

**Review by tgravescs — APPROVED** [review]

File: integration_tests/src/main/python/delta_lake_test.py (hunk 1 above)
Comment (andygrove) [reply integration_tests/src/main/python/delta_lake_test.py:20]: That's correct. This is not needed for the test just helpful to see what is happening.

**Pull request merged**